vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -306,8 +308,127 @@ public IModuleLinkDisplay createModuleLink(String title, ISpace source, float po
         return display;
     }
 
+    @Override
+    public IModuleLinkDisplay editModuleLink(String title, String id, float positionX, float positionY,
```

Comment:
I would refactor here a bit. To avoid this class getting too big, create subclasses for each Link type and use the template pattern to create the methods:
- the superclass has save/edit/delete/etc methods that call abstract methods (implemented in the child classes) that do the specifics.
- e.g. superclass calls `spaceValidation`, `linksValidation` etc and then calls abstract methods that set link properties, etc.